jwst/regtest/test_nirspec_masterbackground.py

```diff
@@ -0,0 +1,174 @@
+from jwst import datamodels
+
+pytestmark = pytest.mark.bigdata
+
+
+@pytest.fixture
+def run_nirspec_mbkg(jail, rtdata_module):
+    """Fixture-factory that runs the MasterBackground step for NIRSpec MasterBackground tests."""
+    rtdata = rtdata_module
+    collect_pipeline_cfgs('config')
+
+    def _set_pipeline_run(input_file: str, user_files: List[tuple] = None, asn: bool = False,
+                          options: List[str] = None):
+        """Set additional pipeline options for the different MasterBackground tests.
+        Additional user_files and their inputs (inputs are assumed to be artifactory paths) should be given as tuples
+        for the user_files argument.
+        Set asn to True if getting an association.
+        Add additional command line options with the options argument.
+        """
+        # Create optional arguments list using rtdata to find artifact paths
+        user_files = [
+            f'{arg}={rtdata.get_data(input_path)}' for arg, input_path in user_files
+        ] if user_files is not None else []
+
+        if options is None:
+            options = []
+
+        # Get the input data
+        if asn:
+            rtdata.get_asn(input_file)
+
+        else:
+            rtdata.get_data(input_file)
+
+        # Set arguments and execute
+        args = ['config/master_background.cfg', rtdata.input, '--save_results=True'] + user_files + options
+        step = Step.from_cmdline(args)
+
+        return rtdata, step
+
+    return _set_pipeline_run
+
+
+def test_nirspec_fs_mbkg_user(run_nirspec_mbkg, fitsdiff_default_kwargs):
+    """Run a test for NIRSpec FS data with a user-supplied background file."""
+    rtdata, _ = run_nirspec_mbkg(
+        'nirspec/fs/nrs_sci+bkg_cal.fits',
+        user_files=[('--user_background', 'nirspec/fs/nrs_bkg_user_clean_x1d.fits')]
+    )
+
+    rtdata.output = "nrs_sci+bkg_master_background.fits"
+
+    # Get the truth file
+    rtdata.get_truth(os.path.join("truth/test_nirspec_fs_mbkg_user", "nrs_sci+bkg_master_background.fits"))
+
+    # Compare the results
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+
+def test_nirspec_ifu_mbkg_user(run_nirspec_mbkg, fitsdiff_default_kwargs):
+    """Test NIRSpec IFU data with a user-supplied background file."""
+    rtdata, step = run_nirspec_mbkg(
+        'nirspec/ifu/prism_sci_bkg_cal.fits',
+        user_files=[('--user_background', 'nirspec/ifu/prism_bkg_x1d.fits')]
+    )
+
+    rtdata.output = 'prism_sci_bkg_master_background.fits'
+
+    # Get the truth file
+    rtdata.get_truth(os.path.join("truth/test_nirspec_ifu_mbkg_user", 'prism_sci_bkg_master_background.fits'))
+
+    # Compare the results
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+    # Test 2  compare the science  data with no background to the output from the masterBackground Subtraction step
```

Comment:
These additional tests that compare data processed one way to data from some other source were originally implemented in the old reg test system when we were in the early stages of developing the master background step and needed some sanity checks to verify that the overall algorithm was doing the right thing (i.e. getting close to the expected result generated by hand). Now that we're confident in the design of the algorithm itself, I think these types of additional tests could be removed. They add a lot of extra baggage and are difficult to maintain and update. All we really need are the very simple tests that run the step and compare its various forms of output products against truth files of the same product.

Reply:
No problem! I can get rid of those